### ISSUE TITLE
App links: page action or menu option to open the current page in an external app

### ISSUE THREAD

**Issue by jhugman** [mozilla-mobile/android-components#3695]
We can already detect if a URL is openable by a user-installed external app. 

Show a page action and menu option to open the current page in that app. 

Bonus: If the user has already selected a default, then show the app icon and name.

**Comment by jhugman** [comment]
@topotropic This is the simplest path forward. Where should this bit of UI go?

**Comment by jhugman** [comment]
@topotropic have talked about this with @shortlander, re: using the existing permission prompts.

> we considered using a similar UI as the one we use for permissions (bottom sheet) --- when you open permission.site with FF Preview and enable camera for example you can see it. Another idea that floated around was something simpler, like when you longpress on a link, open it in the background and see the snackbar that gives you the option to switch to the tab.

TL; DR
1. use the existing permission style prompts
2. use a snackbar.

@shorlander: do you have any preference?

**Comment by jhugman** [comment]
@shorlander : "Can we put this in a Quick Action Bar, perhaps using the app's icon".

Yes, but we'll also a default icon when the user has not selected yet selected an app, and there are two non-browser apps available.

**Comment by jhugman** [comment]
![without app icon](https://user-images.githubusercontent.com/99338/61291762-54876280-a7c7-11e9-9a25-20d348cf3aab.png)
![with app icon](https://user-images.githubusercontent.com/99338/61291786-623ce800-a7c7-11e9-9b11-f7fe1882c37b.png)

**Comment by jhugman** [comment]
![in product without icon](https://user-images.githubusercontent.com/99338/61311927-49e0c380-a7ef-11e9-86ab-49ed6bccedf1.png)

**Comment by abodea** [comment]
| Command | Description |
| --- | --- |
| Whatsapp | List all new or modified files |
| Whatsapp | List all new or modified files |
| Whatsapp | List all new or modified files |

**Comment by abodea** [comment]
Hi @jhugman, I did some testing on the following websites/apps.
I marked them with works/doesn't work meaning that the button `Open in app` appear or not on the website. 
There are few websites that works correctly like your example with Reddit.
I have two questions:
It is fine do keep this bug for tracking all the examples below, or should I add another one in the Fenix repo regarding this issue?
Should we have for all websites a redirection to the play store in case we don't have the app installed?
## For more details please see the information below:

| **Command** | **Description/Status**|
| --- | --- |
| Whatsapp | Need to tap on "Get it on Google Play" resulting a redirect within the browser to the playstore, the "Open in App" is displayed and by pressing it you are redirected to the playstore app. |
| TikTok | Doesn't work |
| Youtube Music | Works |
| Youtube | Works |
| Instagram |  Need to tap on "Get it on Google Play" resulting a redirect within the browser to the playstore, the "Open in App" is displayed and by pressing it you are redirected to the playstore app. |
| Facebook  | Doesn't work |
| Facebook Messenger | Need to tap on "Get it on Google Play" resulting a redirect within the browser to the playstore, the "Open in App" is displayed and by pressing it you are redirected to the playstore app. |
| OLX  | Works |
| Pinterest | Works |
| Netflix | Doesn't work |
| Booking | Doesn't work, from the the lower section of the page "Continue in the app" by tapping "Open App" redirects you to the Playstore app |
| AirBnb | Works |
| Ebay  | List all new or modified files |
| AliExpress | Doesn't work |
| AliBaba | Doesn't work |
| Tumblr | Works  |
